samples: openthread: introduce snippets to openthread samples (#12909)
nrfconnect/sdk-nrf: doc/nrf/protocols/thread/configuring.rst

* :kconfig:option:`CONFIG_OPENTHREAD_DEBUG` - This option enables logging for the OpenThread stack.

Both options must be enabled to allow logging.
Use the ``logging`` snippet to enable both options for the Thread samples in the |NCS|.

After setting these options, you can choose one of several :ref:`logging backends <ug_logging_backends>` available in Zephyr and supported in the |NCS|.

> The :file:`logging.conf` configuration file in the :file:`snippets/logging/` directory of the sample enables :ref:`ug_logging_backends_rtt` as the logging backend by default.

Reviewer: ```suggestion
The ``logging`` snippet enables :ref:`ug_logging_backends_rtt` as the logging backend by default.
```

Author: fixed
